pkg/health/health.go

```diff
@@ -329,6 +329,12 @@ func getStatefulSetHealth(obj *unstructured.Unstructured) (*HealthStatus, error)
 			Message: fmt.Sprintf("partitioned roll out complete: %d new pods have been updated...", sts.Status.UpdatedReplicas),
 		}, nil
 	}
+	if sts.Spec.UpdateStrategy.Type == appsv1.OnDeleteStatefulSetStrategyType {
+		return &HealthStatus{
+			Status:  HealthStatusHealthy,
+			Message: fmt.Sprintf("statefulset has %d ready pods", sts.Status.ReadyReplicas),
```

Comment:
I understand the issue that this change was fixing and it makes sense to me that this counts as "healthy", but what do folks think about iterating on this so that the message lets you know whether or not UpdateRevision matches CurrentRevision in the OnDelete case?

Reply:
https://github.com/argoproj/gitops-engine/pull/575